### ISSUE TITLE
Appimage 3.0.2 not starting with user login or with restart and login

### ISSUE THREAD

**Issue by Steve-Z** [nextcloud/desktop#2504]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Expected behaviour
After selecting Settings > General > General Settings and checking "Launch on system startup", I expect the appimage client to start itself when I login as user with admin privileges and owner of appimage file.

### Actual behaviour
I can start the appimage manually by double-clicking the file, but if I log out and then log back in, the client does not start automatically.

### Steps to reproduce
1. Login and manually start client. Sync is successful.
2. Make sure "Launch on system startup is checked" in General Settings
3. Logout
4. Log back in to see that client has not started.
5. Reboot and login has same result.

### Client configuration
Client version:  Nextcloud-3.0.2-x86_64.Appimage

Operating system: Pop_OS 20.04

OS language: Linux

Qt version used by client package (Linux only, see also Settings dialog):

Client package (From Nextcloud or distro) (Linux only): From Nextcloud

Installation path of client: $HOME/Appimages/

### Server configuration

Nextcloud version: 19.0.3.1

Storage backend (external storage): None

### Logs

Please use Gist (https://gist.github.com/) or a similar code paster for longer
logs.

1. Client logfile: 

Output of `nextcloud --logdebug --logwindow` or `nextcloud --logdebug --logfile log.txt`
(On Windows using `cmd.exe`, you might need to first `cd` into the Nextcloud directory)
(See also https://docs.nextcloud.com/desktop/2.3/troubleshooting.html#log-files)

When I type those commands, response asks if I would like to `sudo apt install nextcloud-desktop`. I think this is because I am using an appimage.

2. Web server error log: n/a

3. Server logfile: nextcloud log (data/nextcloud.log): n/a

**UPDATE**:
I changed "Launch on system startup" setting off and on. I looked to see which files changed. From `$HOME` I ran `find . -cmin 1` and discovered this: 
`~/.config/autostart/com.nextcloud.desktopclient.nextcloud.desktop`

Here is that file:
```
[Desktop Entry]
Name=Nextcloud
GenericName=File Synchronizer
Exec=/tmp/.mount_NextclqpeKeD/usr/bin/nextcloud --background
Terminal=false
Icon=Nextcloud
Categories=Network
Type=Application
StartupNotify=false
X-GNOME-Autostart-enabled=true
```
Interestingly the `/tmp` directory shows a slightly different `.mount_Nextclxxxxxx/`. This puzzled me at first, but then I realized that manually starting the appimage client will create a new `nextcloud` executable path in the `/tmp` directory. That newly mounted directory will not match the path in the non-functioning `*.desktop` file. 

Bottom line: `autostart` is not functioning. Is this specific to GNOME?

**Comment by Steve-Z** [comment]
I found a workaround to this. The problem is this setting in 
`~/.config/autostart/com.nextcloud.desktopclient.nextcloud.desktop`:
```
...
Exec=/tmp/.mount_NextclqpeKeD/usr/bin/nextcloud --background
...
```
I changed that using the _Startup Applications Preferences_ utility in Pop_OS applications [super-A] > System group. I edited the "Command" field to `path/to/Nextcloud-3.0.2-x86_64.AppImage`. I assume Ubuntu would be similar.

That utility changed the`Exec` value to that same path. I assume you can edit this directly with `nano ~/.config/autostart/com.nextcloud.desktopclient.nextcloud.desktop` (or editor of your choice).

I am not going to close the issue. I think the AppImage developers should fix this small bug so that the `Exec` value is populated correctly when "Launch on system startup" is checked.

**Comment by er-vin** [comment]
Yes indeed, the nature of the appimage makes it more complicated to generate a proper autostart entry. As got noted here the path to the running executable changes. We'd need a way to catch that we're running from an appimage and use the path to said appimage when registering the binary in the autostart.

I don't know if there's a way for appimage applications to do this, this would require some investigation.

**Comment by thomasboehm** [comment]
A permanent workaround is to start the AppImage once and then copy the content of the whole directory /tmp/.mount_NextclqpeKeD somewhere else (/opt or somewhere in your home directory) and adjust the Exec line in the autostart file accordingly.

**Comment by hrittich** [comment]
It should be possible for an AppImage-packed application, to create a proper autostart `.desktop` entry. As has been discussed here, the NextCloud client is able to create the required `.desktop` entry, however, sets the wrong application path. It sets the path of the application in the temporary mount point created by AppImage, instead of setting the path to the AppImage itself. Within the AppImage runtime, the path to the AppImage is provided via the `APPIMAGE` environment variable (See [AppImage Environment Variables]). Hence, setting the correct path should be possible.

[AppImage Environment Variables]: https://docs.appimage.org/packaging-guide/environment-variables.html?highlight=type#type-2-appimage-runtime

**Comment by er-vin** [comment]
Thanks for the information, should make for a small patch indeed. Any volunteer? :-)

**Comment by PatrickJRed** [comment]
I had this Issue as well,
changing to the path of the appimage-file helped me.
 for : "create a proper autostart .desktop entry" the file could contain this for exec "Exec=xdg-open PATH_TO_AppImage" (works for me)

**Comment by hrittich** [comment]
Anyone having this issue can try the AppImage in #2739 to see whether it fixes the problem.
